internal/clients/clients.go

```diff
@@ -0,0 +1,108 @@
+package clients
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/selection"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/tools/clientcmd"
+	clusterv1 "open-cluster-management.io/api/cluster/v1"
+	clusterv1beta1 "open-cluster-management.io/api/cluster/v1beta1"
+	clusterv1beta2 "open-cluster-management.io/api/cluster/v1beta2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+)
+
+const (
+	ListRequestTimeout   = 30 * time.Second
+	SingleRequestTimeout = 10 * time.Second
+
+	secretTypeLabel      = "hive.openshift.io/secret-type"
+	secretTypeLabelValue = "kubeconfig"
```

Comment:
could add a comment about where these are coming from just to make it clear for the future? 

Also are the timeouts related to these two consts if not maybe create new const block (nit)?